WallPaperDev/WallPaperDev/Views/Controllers/SearchImageViewController.swift

```diff
@@ -0,0 +1,147 @@
+    NSLayoutConstraint.activate([
+      searchImagesCV.leadingAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.leadingAnchor),
+      searchImagesCV.trailingAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.trailingAnchor),
+      searchImagesCV.topAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.topAnchor),
+      searchImagesCV.bottomAnchor.constraint(equalTo: self.view.bottomAnchor)
+    ])
+  }
+  
+  private func setupNavBar() {
+    searchController.searchBar.delegate = self
+    searchController.searchBar.placeholder = "Search for images"
+    searchController.searchBar.autocorrectionType = .default
+    searchController.searchBar.sizeToFit()
+    
+    navigationItem.title = "Search for a new image"
+    navigationController?.navigationBar.largeTitleTextAttributes = navigationController?.navigationBar.configLargeText(length: "Search Images")
+    
+    navigationItem.hidesSearchBarWhenScrolling = false
+    navigationItem.searchController = searchController
+  }
+  
+  private func setupBlueButton() {
+    self.view.addSubview(createImageButton)
+    createImageButton.setTitle("Create", for: .normal)
+    createImageButton.addTarget(self, action: #selector(backToChooseImageVC), for: .touchUpInside)
+    NSLayoutConstraint.activate([
+      createImageButton.widthAnchor.constraint(equalTo: self.view.widthAnchor, multiplier: 0.6),
+      createImageButton.heightAnchor.constraint(equalToConstant: 50),
+      createImageButton.bottomAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.bottomAnchor),
+      createImageButton.centerXAnchor.constraint(equalTo: self.view.centerXAnchor)
+    ])
+  }
+  
+  @objc private func backToChooseImageVC() {
+    if let image = viewModel.selectedImage {
+      selectedImageDelegate?.passImageSelected(image: image)
+    }
+    navigationController?.popViewController(animated: true)
+  }
+}
+
+// MARK: - CollectionView Delegate
+extension SearchImageViewController : UICollectionViewDelegate {
+  func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+    guard let selectedCell = searchImagesCV.cellForItem(at: indexPath) as? SearchImagesCell else {
+      return
+    }
+    viewModel.selectedImage = selectedCell.cellImage
+  }
+}
+
+// MARK: - CollectionView DataSource
+extension SearchImageViewController : UICollectionViewDataSource {
+  func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+    viewModel.imageURLS.count
+  }
+
+  func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+    guard let cell = searchImagesCV.dequeueReusableCell(withReuseIdentifier: searchImagesCV.cellID, for: indexPath) as? SearchImagesCell else {
+      return UICollectionViewCell()
+    }
+
+    let urlString = viewModel.imageURLS[indexPath.row].urls.regular
+
+    viewModel.networkManager.getPhotoData(from: urlString) { result in
```

Comment:
Not a huge concern right now, but as cells are being recycled we will not kill these previous tasks therefore we will be wasting some resources and experience flickering